### ISSUE TITLE
Difference between `nll_logistic_hazard` and the one from Gensheimer (2018) ?

### ISSUE THREAD

**Issue by peterchristofferholm** [havakv/pycox#129]
Hey, thanks for an awesome package!

In the implementation of the loss function presented in [1] , the log-likelihood for a **censored** individual is presented as

![image](https://user-images.githubusercontent.com/23274836/162732416-d6c2686f-961f-41f5-9e66-935acdd81338.png)

which means that an individual only contributes to the loss up and until the time of censoring. As far as I can tell, in the implementation you've provided, individuals still contribute to the loss after they're censored. My questions are:

1.  Is that correctly understood? Is it intentional?

2. If (1) wouldn't that bias the prediction towards survival since censored individuals are incorrectly treated as "survivors"?

3. If not (1), how are the censored individuals "removed" from the log-likelihood?

[1] Gensheimer, Michael F., and Balasubramanian Narasimhan. "A scalable discrete-time survival model for neural networks." PeerJ 7 (2019): e6257.

**Comment by havakv** [comment]
Hei, and thank you for the kind words.

1. The individuals should not contribute to the loss after their event or censoring time. We describe how this is done in Equation (7) of [our paper](https://link.springer.com/article/10.1007/s10985-021-09532-6#Equ7) where the seconds sum (over j representing time) stops at the even/censoring time.
2. Does not apply, but yes that is correct.
3. In [our code](https://github.com/havakv/pycox/blob/master/pycox/models/loss.py#L50) this is achieved using the pytorch gather function.

Hope this clears things up

**Comment by peterchristofferholm** [comment]
1. Ahh, I didn't realize that $\kappa(\tau_i)$ are of course dependent on the individual. Yeah, that makes sense.
2. 👍 
3. Cool use of `gather`. I think I prefer that approach than the clunky survarray think they're using in the Gensheimer implenentation. Might steal it! 

I'm not that good with the mathy-sections of ML papers, so that's probably the reason I got confused. I'm glad that you took the time to respond. That definitely cleared things up.
